### ISSUE TITLE
Paket pack: need a way to specify current nuget version as dependency

### ISSUE THREAD

**Issue by eiriktsarpalis** [fsprojects/Paket#837]
I'm interested in integrating the `paket pack` functionality in my projects, so I've started playing around with it in FsPickler. In FsPickler there are two projects, FsPickler and FsPickler.Json. FsPickler has no nuget dependencies and FsPickler.Json depends on 1) Json.Net and 2) FsPickler of precisely the same version number. 

I can rely on the Json.Net dependency being added automatically by `paket pack`, however I can't find a way to specify the latter dependency correctly. Is there any way I can achieve this through the `paket.template` syntax or FAKE?

**Comment by forki** [comment]
Mhm good question. @mavnn do you have a suggestion how we could implement such a feature?

**Comment by forki** [comment]
@eiriktsarpalis how would you like to configure this feature. I think I can add it easily, but we need to define where and how.

**Comment by eiriktsarpalis** [comment]
I could think if two possible ways: either use a macro in the template
syntax for the current version or have FAKE pass additional dependencies to
the pack configuration record.

**Comment by forki** [comment]
I think it should be done in the template.

**Comment by eiriktsarpalis** [comment]
I tend to agree, it would make package definitions more self contained.

**Comment by forki** [comment]
do you already started the conversion? I'd love to use it as a sample to implement this.

**Comment by eiriktsarpalis** [comment]
Yeah, I started an effort in this [branch](https://github.com/nessos/FsPickler/tree/paket-pack), though I never completed it.

**Comment by eiriktsarpalis** [comment]
I'm getting the following error while trying to pack that FsPickler branch with 1.9.0:
```
eirik@FELIXPC ~/Development/nessos/FsPickler (paket-pack)
$ .paket/paket.exe pack output bin
Paket version 1.9.0.0

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Paket.TemplateFileModule.f@1-85(FSharpOption`1 currentVersion, String d)
   at Paket.TemplateFileModule.getDependencies@206-1.Invoke(String[] array)
   at Paket.TemplateFileModule.getDependencies(FSharpMap`2 map, FSharpOption`1 currentVersion)
   at Paket.TemplateFileModule.getOptionalInfo(FSharpMap`2 map, FSharpOption`1 currentVersion)
   at Paket.TemplateFileModule.Parse@313-9.Invoke(PackageConfigType _arg3)
   at Chessie.ErrorHandling.Trial.fSuccess@63-11[p,r,q](FSharpFunc`2 f, p x, FSharpList`1 msgs)
   at Chessie.ErrorHandling.Trial.fSuccess@63-11[p,r,q](FSharpFunc`2 f, p x, FSharpList`1 msgs)
   at Chessie.ErrorHandling.Trial.TrialBuilder.Using[d,e](d d, FSharpFunc`2 body)
   at Paket.TemplateFileModule.Load(String fileName, FSharpOption`1 currentVersion)
   at Paket.PackageProcess.projectTemplates@33-1.Invoke(ProjectFile projectFile)
   at Microsoft.FSharp.Collections.ArrayModule.Choose[T,TResult](FSharpFunc`2 chooser, T[] array)
   at Paket.PackageProcess.Pack(DependenciesFile dependencies, String packageOutputPath, FSharpOption`1 buildConfig, FSharpOption`1 version, FSharpOption`1 releaseNotes, FSharpOption`1 templateFile)
   at Paket.Program.clo@279-26.Invoke(ArgParseResults`1 results)
   at Paket.Program.processWithValidation[T](FSharpFunc`2 validateF, FSharpFunc`2 commandF, Command command, String[] args)
   at Paket.Program.processCommand@70-1.Invoke(Command command, String[] args)
   at <StartupCode$paket>.$Paket.Program.main@()
```

**Comment by forki** [comment]
thx. I found this bug and fixed it in 1.9.1.
it now works on my machine, but only after I renamed your "Paket.template" files to "paket.template" (all lowercase)

**Comment by eiriktsarpalis** [comment]
Thanks, seems to be working now.
